src/main/webapp/script.js

```diff
@@ -387,12 +388,19 @@ function getWeeklyView(aggregated) {
   thirtyDaysFromNow.setDate(thirtyDaysFromNow.getDate() - 30);
 
   var firstDayOfWeek = new Date(thirtyDaysFromNow);
-  firstDayOfWeek.setDate(firstDayOfWeek.getDate() - (today.getDay() - 1));
+  console.log(firstDayOfWeek);
+  console.log(firstDayOfWeek.getDay());
+  if (firstDayOfWeek.getDay() == 0) {
```

Comment:
In Javascript you should use === instead of == because == checks if equality, disregarding the variables types. For exampla 2 == '2' would be true. I would change it  to 
`if (firstDayOfWeek.getDay() === 0) {`

Reply:
More here [Difference between === and == in JavaScript](https://www.geeksforgeeks.org/difference-between-and-operator-in-javascript/)

Reply:
Ooh good point - I hadn't come across === before! Thanks for the extra info as well :D

Reply:
Fixed in 4b1a871.